app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java

```diff
@@ -0,0 +1,104 @@
+    @Override protected void onResume() {
+        super.onResume();
+
+        myListView = findViewById(R.id.myAccountLayout);
+        initializeArrays();
+
+        ArrayList<MyAccountButton> myFilteredAccountButtons = filterButtonList(
+                myAccountButtonList, (b -> (b.isActiveForPlayer() == userProfile.getIsPlayer() ||
+                        b.isActiveForGM() == userProfile.getIsGm())));
+        List<Drawable> myFilteredDrawables = cutDownImagesList();
+
+        myAdapter = new MyAccountButtonAdapter(this, myFilteredAccountButtons, myFilteredDrawables);
+        myListView.setAdapter(myAdapter);
+        myListView.setOnItemClickListener((l, v, position, id) -> startActivity(
+                new Intent(this, myFilteredAccountButtons.get(position).getActivityClass())));
+
+    }
+
+    private void initializeArrays() {
+        myAccountButtonList = Arrays.asList(
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.pendingRequestText), PendingRequestActivity.class, true, false),
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.confirmedGamesText), ConfirmedGamesActivity.class, true, false),
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.pastGamesText), PastGamesActivity.class, true, false),
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.hostedGamesText), HostedGamesActivity.class, false, true),
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.pastHostedGamesText), PastHostedGamesActivity.class, false, true),
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.profileText), ProfileActivity.class, true, true)
+        );
+        myDrawablesList = Arrays.asList(
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name)
+        );
+    }
+
+    private List<Drawable> cutDownImagesList() {
+        ArrayList<Drawable> myFilterDrawableList = new ArrayList<>(myDrawablesList);
```

Comment:
Can you please use something more functionnal?
- Make a stream out of account buttons (or text for that matter)
- Make a stream with all the images
- Use `com.annimon.stream.Stream.zip` to make a stream of pairs.
- Collect the resulting stream into a list of pairs
- Replace the image on the last element of the pair (since it corresponds to the profile)